intbot/core/integrations/github.py

```diff
@@ -55,127 +77,175 @@ def as_discord_message(self):
         return f"[{self.title}]({self.url})"
 
 
-@dataclasses.dataclass
-class GithubDraftIssue:
+class GithubDraftIssue(BaseModel):
     id: str
     title: str
 
     def as_discord_message(self):
         return self.title
 
 
-def fetch_github_item_details(item_id):
-    headers = {
-        "Authorization": f"Bearer {settings.GITHUB_API_TOKEN}",
-        "Content-Type": "application/json",
-    }
-    payload = {"query": query, "variables": {"itemId": item_id}}
-    response = httpx.post(GITHUB_API_URL, json=payload, headers=headers)
-    if response.status_code == 200:
-        return response.json()["data"]["node"]["content"]
-    else:
-        raise Exception(f"GitHub API error: {response.status_code} - {response.text}")
+class GithubWebhook:
+    """
+    Base class for all the other specific types of webhooks.
+    """
 
+    def __init__(self, action: str, headers: dict, content: dict, extra: dict):
```

Comment:
Can we have for dict full types? i.e. `dict[str, str]` or such?

Reply:
Changing to JsonType for now - since all the webhooks have a slightly different shape.